models/factor.go

```diff
@@ -12,7 +12,55 @@ import (
 const FactorUnverifiedState = "unverified"
 const FactorVerifiedState = "verified"
 
-const TOTP = "TOTP"
+type AuthenticationMethod int
+
+const (
+	OAuth AuthenticationMethod = iota
+	OAuthIDGrant
+	PasswordGrant
+	AutoConfirmSignup
+	EmailVerification
+	SMSOrEmailOTP
+	TOTP
+)
+
+func (authMethod AuthenticationMethod) String() string {
+	switch authMethod {
+	case OAuth:
+		return "oauth"
+	case OAuthIDGrant:
+		return "oauth_id"
+	case PasswordGrant:
+		return "password"
+	case AutoConfirmSignup:
+		return "autoconfirm"
+	case EmailVerification:
+		return "email_verification"
+	case SMSOrEmailOTP:
+		return "sms_or_email_otp"
+	case TOTP:
+		return "TOTP"
+	default:
+		return ""
```

Comment:
You can just remove this, the Go compiler is smart enough when it comes to `iota`s.

Reply:
```suggestion
```